app/routes/user_routes.py

```
@@ -91,14 +91,14 @@ def delete_user(user_id: int, db: Session = Depends(get_db)):
 # Update a user's role (admin only)
 # -----------------------------
 @router.patch("/{user_id}/role")
-def update_user_role(user_id: int, role: UserRole, db: Session = Depends(get_db)):
+def update_user_role(user_id: int, payload: UpdateUserRole, db: Session = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)